src/udev/memory_id/types.h

```diff
@@ -0,0 +1,63 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#pragma once
+
+#include <inttypes.h>
+#include "dmidecode-config.h"
+
+struct dmi_header
+{
+        uint8_t type;
+        uint8_t length;
+        uint16_t handle;
+        uint8_t *data;
+};
+
+/*
+ * You may use the following defines to adjust the type definitions
+ * depending on the architecture:
+ * - Define ALIGNMENT_WORKAROUND if your system doesn't support
+ *   non-aligned memory access. In this case, we use a slower, but safer,
+ *   memory access method. This should be done automatically in config.h
+ *   for architectures which need it.
+ */
+
+#include <endian.h>
+
+#if __BYTE_ORDER == __BIG_ENDIAN
+typedef struct {
+        uint32_t h;
+        uint32_t l;
+} u64;
```

Comment:
the kernel has a type called "u64", and it's not really defined like this. i'd rather we name this differently to avoid confusion. At least to me it's a first time I see a type called "u64" defined as struct...

Reply:
Called it `ULargeInteger`, inspired by https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-ularge_integer-r1